File: Android Project/app/src/main/java/com/example/histoquiz/util/RoomCreator.java
package com.example.histoquiz.util;

import java.util.Random;

public class RoomCreator {
    protected String upperAlphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    protected String lowerAlphabet = "abcdefghijklmnopqrstuvwxyz";
    protected String numbers = "0123456789";
    protected String alphaNumeric = upperAlphabet + lowerAlphabet + numbers;
    protected StringBuilder sb = new StringBuilder();
    protected Random random = new Random();
    protected String actualRoomName;

    public String newRoomCode(int length){
        for(int i = 0; i < length; i++) {
            int index = random.nextInt(alphaNumeric.length());
            char randomChar = alphaNumeric.charAt(index);
            sb.append(randomChar);
        }
        actualRoomName = sb.toString();
        return actualRoomName;
    }

    public String getActualRoomName(){
        return actualRoomName;
    }

    public void setActualRoomName(String roomName){
        this.actualRoomName = roomName;
    }
}

File: Android Project/app/src/main/java/com/lenda/histoquiz/util/RoomCreator.kt
package com.lenda.histoquiz.util

import java.util.*

class RoomCreator {
    var lowerAlphabet = "abcdefghijklmnopqrstuvwxyz"
    var actualRoomName: String? = null
    var numbers = "0123456789"
    private var upperAlphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
    private var alphaNumeric = upperAlphabet + lowerAlphabet + numbers
    private var sb = StringBuilder()
    private var random = Random()

    fun newRoomCode(length: Int): String {
        for (i in 0 until length) {
            val index = random.nextInt(alphaNumeric.length)
            val randomChar = alphaNumeric[index]
            sb.append(randomChar)
        }
        actualRoomName = sb.toString()
        return actualRoomName as String
    }
}